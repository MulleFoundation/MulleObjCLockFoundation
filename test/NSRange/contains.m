#import <MulleObjC/MulleObjC.h>

#include <stdio.h>
#include <stdint.h>



static void   test_contains_range( NSRange range1, NSRange range2)
{
   printf( "[%ld,%ld] contains range [%ld,%ld] ? %s\n",
               (long) range1.location, (long) range1.length,
               (long) range2.location, (long) range2.length,
      MulleObjCRangeContainsRange( range1, range2) ? "YES" : "NO");
}


int   main( void)
{
   NSRange   range1;
   NSRange   range2;

   for( range2.location = -2; range2.location != 3; range2.location++)
      for( range2.length = -2; range2.length != 3; range2.length++)
         if( MulleObjCRangeIsValid( range2))
         {

            for( range1.location = -2; range1.location != 3; range1.location++)
               for( range1.length = -2; range1.length != 3; range1.length++)
                  if( MulleObjCRangeIsValid( range1))
                     test_contains_range( range1, range2);
            printf( "\n");
         }
   return( 0);
}

