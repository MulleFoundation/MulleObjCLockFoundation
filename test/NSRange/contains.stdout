[0,0] contains range [0,0] ? NO
[0,1] contains range [0,0] ? NO
[0,2] contains range [0,0] ? NO
[1,-1] contains range [0,0] ? NO
[1,0] contains range [0,0] ? NO
[1,1] contains range [0,0] ? NO
[1,2] contains range [0,0] ? NO
[2,-2] contains range [0,0] ? NO
[2,-1] contains range [0,0] ? NO
[2,0] contains range [0,0] ? NO
[2,1] contains range [0,0] ? NO
[2,2] contains range [0,0] ? NO

[0,0] contains range [0,1] ? NO
[0,1] contains range [0,1] ? YES
[0,2] contains range [0,1] ? YES
[1,-1] contains range [0,1] ? NO
[1,0] contains range [0,1] ? NO
[1,1] contains range [0,1] ? NO
[1,2] contains range [0,1] ? NO
[2,-2] contains range [0,1] ? NO
[2,-1] contains range [0,1] ? YES
[2,0] contains range [0,1] ? NO
[2,1] contains range [0,1] ? NO
[2,2] contains range [0,1] ? NO

[0,0] contains range [0,2] ? NO
[0,1] contains range [0,2] ? NO
[0,2] contains range [0,2] ? YES
[1,-1] contains range [0,2] ? NO
[1,0] contains range [0,2] ? NO
[1,1] contains range [0,2] ? NO
[1,2] contains range [0,2] ? NO
[2,-2] contains range [0,2] ? NO
[2,-1] contains range [0,2] ? NO
[2,0] contains range [0,2] ? NO
[2,1] contains range [0,2] ? NO
[2,2] contains range [0,2] ? NO

[0,0] contains range [1,-1] ? NO
[0,1] contains range [1,-1] ? NO
[0,2] contains range [1,-1] ? NO
[1,-1] contains range [1,-1] ? YES
[1,0] contains range [1,-1] ? NO
[1,1] contains range [1,-1] ? NO
[1,2] contains range [1,-1] ? NO
[2,-2] contains range [1,-1] ? NO
[2,-1] contains range [1,-1] ? NO
[2,0] contains range [1,-1] ? NO
[2,1] contains range [1,-1] ? NO
[2,2] contains range [1,-1] ? NO

[0,0] contains range [1,0] ? NO
[0,1] contains range [1,0] ? NO
[0,2] contains range [1,0] ? NO
[1,-1] contains range [1,0] ? NO
[1,0] contains range [1,0] ? NO
[1,1] contains range [1,0] ? NO
[1,2] contains range [1,0] ? NO
[2,-2] contains range [1,0] ? NO
[2,-1] contains range [1,0] ? NO
[2,0] contains range [1,0] ? NO
[2,1] contains range [1,0] ? NO
[2,2] contains range [1,0] ? NO

[0,0] contains range [1,1] ? NO
[0,1] contains range [1,1] ? NO
[0,2] contains range [1,1] ? YES
[1,-1] contains range [1,1] ? YES
[1,0] contains range [1,1] ? NO
[1,1] contains range [1,1] ? YES
[1,2] contains range [1,1] ? YES
[2,-2] contains range [1,1] ? NO
[2,-1] contains range [1,1] ? NO
[2,0] contains range [1,1] ? NO
[2,1] contains range [1,1] ? NO
[2,2] contains range [1,1] ? NO

[0,0] contains range [1,2] ? NO
[0,1] contains range [1,2] ? NO
[0,2] contains range [1,2] ? NO
[1,-1] contains range [1,2] ? YES
[1,0] contains range [1,2] ? NO
[1,1] contains range [1,2] ? NO
[1,2] contains range [1,2] ? YES
[2,-2] contains range [1,2] ? NO
[2,-1] contains range [1,2] ? NO
[2,0] contains range [1,2] ? NO
[2,1] contains range [1,2] ? NO
[2,2] contains range [1,2] ? NO

[0,0] contains range [2,-2] ? NO
[0,1] contains range [2,-2] ? NO
[0,2] contains range [2,-2] ? NO
[1,-1] contains range [2,-2] ? YES
[1,0] contains range [2,-2] ? NO
[1,1] contains range [2,-2] ? NO
[1,2] contains range [2,-2] ? NO
[2,-2] contains range [2,-2] ? YES
[2,-1] contains range [2,-2] ? YES
[2,0] contains range [2,-2] ? NO
[2,1] contains range [2,-2] ? NO
[2,2] contains range [2,-2] ? NO

[0,0] contains range [2,-1] ? NO
[0,1] contains range [2,-1] ? NO
[0,2] contains range [2,-1] ? NO
[1,-1] contains range [2,-1] ? NO
[1,0] contains range [2,-1] ? NO
[1,1] contains range [2,-1] ? NO
[1,2] contains range [2,-1] ? NO
[2,-2] contains range [2,-1] ? NO
[2,-1] contains range [2,-1] ? YES
[2,0] contains range [2,-1] ? NO
[2,1] contains range [2,-1] ? NO
[2,2] contains range [2,-1] ? NO

[0,0] contains range [2,0] ? NO
[0,1] contains range [2,0] ? NO
[0,2] contains range [2,0] ? NO
[1,-1] contains range [2,0] ? NO
[1,0] contains range [2,0] ? NO
[1,1] contains range [2,0] ? NO
[1,2] contains range [2,0] ? NO
[2,-2] contains range [2,0] ? NO
[2,-1] contains range [2,0] ? NO
[2,0] contains range [2,0] ? NO
[2,1] contains range [2,0] ? NO
[2,2] contains range [2,0] ? NO

[0,0] contains range [2,1] ? NO
[0,1] contains range [2,1] ? NO
[0,2] contains range [2,1] ? NO
[1,-1] contains range [2,1] ? YES
[1,0] contains range [2,1] ? NO
[1,1] contains range [2,1] ? NO
[1,2] contains range [2,1] ? YES
[2,-2] contains range [2,1] ? YES
[2,-1] contains range [2,1] ? YES
[2,0] contains range [2,1] ? NO
[2,1] contains range [2,1] ? YES
[2,2] contains range [2,1] ? YES

[0,0] contains range [2,2] ? NO
[0,1] contains range [2,2] ? NO
[0,2] contains range [2,2] ? NO
[1,-1] contains range [2,2] ? YES
[1,0] contains range [2,2] ? NO
[1,1] contains range [2,2] ? NO
[1,2] contains range [2,2] ? NO
[2,-2] contains range [2,2] ? YES
[2,-1] contains range [2,2] ? YES
[2,0] contains range [2,2] ? NO
[2,1] contains range [2,2] ? NO
[2,2] contains range [2,2] ? YES

